vernac/vernacentries.ml

```diff
@@ -2717,16 +2735,30 @@ let interp ?(verbosely=true) ~st cmd =
     Vernacstate.invalidate_cache ();
     iraise exn
 
-let interp_qed_delayed_proof ~proof ~info ~st ?loc pe : Vernacstate.t =
+(* Interpreting a possibly delayed proof *)
+let interp_qed_delayed ~proof ~info ~st pe : Vernacstate.LemmaStack.t option =
   let stack = st.Vernacstate.lemmas in
   let stack = Option.cata (fun stack -> snd @@ Vernacstate.LemmaStack.pop stack) None stack in
+  let () = match pe with
+    | Admitted ->
+      save_lemma_admitted_delayed ~proof ~info
+    | Proved (_,idopt) ->
+      save_lemma_proved_delayed ~proof ~info ~idopt in
+  stack
+
+let interp_qed_delayed_control ~proof ~info ~st ?loc ~control pe : Vernacstate.t =
+  let vernac = CAst.make ?loc { control; attrs = []; expr = VernacEndProof pe } in
+  let stack =
+    List.fold_right (fun flag fn -> interp_control_flag ~vernac flag fn)
+      control
+      (fun ~st -> interp_qed_delayed ~proof ~info ~st pe)
+      ~st in
```

Comment:
Is it just me who likes to have the `in` on its own line for multiline letins?

Reply:
Fixed, but indeed achieving consistency on this stuff is hard, we should switch to `ocamlformat` at some point.